### PR TITLE
feat: add tally/newline-between-instructions rule

### PR DESCRIPTION
**Pull request by tinovyatkin**
## Summary

- Add new `tally/newline-between-instructions` style rule that controls blank lines between Dockerfile instructions
- Three modes: **grouped** (default), **always**, and **never**
- Enabled by default with `FixSafe` auto-fix (priority 55)
- Includes `resolveNodeEndLine()` helper that correctly handles heredoc instructions (uses `Node.EndLine` instead of `Location().End.Line`)

## Details

| Decision | Choice |
|----------|--------|
| Default mode | `grouped` — same-type instructions adjacent, different types separated by blank line |
| Severity | `SeverityStyle` |
| Enabled by default | Yes (`IsExperimental: false`) |
| "always" with multiple blanks | No violation (at least one blank present) |
| "never" with multiple blanks | Single edit removes all |
| Comments between instructions | Belong to next instruction (BuildKit `PrevComment`) |
| Config supports string shorthand | Yes (`"grouped"` / `"always"` / `"never"`) |

## Test plan

- [x] Unit tests: metadata snapshot, default config, config validation, 19 table-driven Check cases (all modes + edge cases), fix safety/edit count tests
- [x] Integration lint test with snapshot
- [x] Integration fix tests for all three modes with snapshots
- [x] Updated total-rules-enabled fixture and snapshot (rules_enabled: 54 → 55)
- [x] Updated LSP test fixtures and snapshots
- [x] `make test` — 3052 tests pass, 0 failures
- [x] `make lint` — 0 issues
- [x] `make cpd` — clean
- [x] `make jsonschema` — schema regenerated

🤖 Generated with [Claude Code](https://claude.com/claude-code)

## Summary by CodeRabbit

* **New Features**
  * Added a style rule "newline-between-instructions" (modes: grouped default, always, never) with auto-fix and editor/CLI fix support.
  * Total enabled rules count increased.

* **Documentation**
  * Added full rule docs with examples, configuration options, and usage guidance.

* **Tests**
  * Added unit/integration tests and updated snapshots, LSP diagnostics, and code-action expectations.

* **Chores**
  * Updated schema and configuration mappings; added indirect module dependencies.

### CODE REVIEW

File: internal/integration/__snapshots__/TestFixPreferAddUnpackBeatsHeredoc_1.snap.Dockerfile [hunk 1]
```diff
@@ -1,7 +1,8 @@
 FROM ubuntu:22.04
 ADD --unpack https://go.dev/dl/go1.22.0.linux-amd64.tar.gz /usr/local
 ADD --unpack https://nodejs.org/dist/v20.11.0/node-v20.11.0-linux-x64.tar.xz /usr/local
-ADD --unpack https://example.com/app.tar.gz /opt
+ADD --unpack https://example.com/app.tar.gz /opttar -xz -C /usr/local
```
[the remark this answers is not in the document]
Comment (tinovyatkin) [reply internal/integration/__snapshots__/TestFixPreferAddUnpackBeatsHeredoc_1.snap.Dockerfile]: ✅ Addressed in 1553317. The `/opttar` corruption was caused by the sync fix approach (HEAD~1) — converting to async resolver in this commit fixed it. The snapshot now correctly shows `/opt`. Thanks @gemini-code-assist\!

File: internal/rules/tally/newline_between_instructions_test.go [hunk 2]
```diff
@@ -0,0 +1,257 @@
...
+			Name:           "global ARG before FROM - grouped missing blank",
+			Content:        "ARG BASE=alpine\nFROM ${BASE}\n",
+			WantViolations: 1,
+			WantMessages:   []string{"expected blank line between ARG and FROM"},
+		},
+		{
+			Name:           "comments between instructions",
+			Content:        "FROM alpine:3.20\n\n# Install dependencies\nRUN echo hello\n",
+			WantViolations: 0,
+		},
```
[the remark this answers is not in the document]
Comment (tinovyatkin) [reply internal/rules/tally/newline_between_instructions_test.go]: ✅ Addressed in 5a0657d. Thanks @coderabbitai\!
